plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -520,30 +533,29 @@ public void abort()
     private static class FileDeletion
     {
         private final List<String> partitionValues;
-        private final LongBitmapDataProvider rowsToDelete = new Roaring64Bitmap();
-        private final String cdfOperation;
+        private final LongBitmapDataProvider rowsDeletedBecauseOfDelete = new Roaring64Bitmap();
+        private final LongBitmapDataProvider rowsDeletedBecauseOfUpdate = new Roaring64Bitmap();
 
-        private FileDeletion(List<String> partitionValues, String cdfOperation)
+        private FileDeletion(List<String> partitionValues)
         {
             // Use ArrayList since Delta Lake allows NULL partition values, and wrap it in
             // an unmodifiableList.
             this.partitionValues = unmodifiableList(new ArrayList<>(requireNonNull(partitionValues, "partitionValues is null")));
-            this.cdfOperation = requireNonNull(cdfOperation, "cdfOperation is null");
         }
 
         public List<String> partitionValues()
         {
             return partitionValues;
         }
 
-        public LongBitmapDataProvider rowsToDelete()
+        public LongBitmapDataProvider rowsDeletedBecauseOfDelete()
         {
-            return rowsToDelete;
+            return rowsDeletedBecauseOfDelete;
         }
 
-        public String getCdfOperation()
+        public LongBitmapDataProvider rowsDeletedBecauseOfUpdate()
         {
-            return cdfOperation;
+            return rowsDeletedBecauseOfUpdate;
```

Comment:
nit: How about renaming to like below? 
* `rowsDeletedBecauseOfDelete` → `rowsDeletedByDelete`
* `rowsDeletedBecauseOfUpdate` → `rowsDeletedByUpdate`